lib/lbt/bundle/Builder.js

```diff
@@ -527,29 +693,90 @@ function rewriteDefine(targetBundleFormat, code, moduleName) {
 			changes.push({
 				// asterisk marks the index: sap.ui.*define()
 				index: defineCall.callee.property.range[0],
-				count: 0,
 				value: "pre"
 			});
 		}
 
-		return applyChanges(codeStr, changes);
+		if (moduleSourceMap === null) {
+			log.info(`No input source map available for module ${moduleName}`);
+		}
+		return transform(moduleContent, changes, moduleSourceMap);
 	}
 
-	return false;
+	return {};
 }
 
-function applyChanges(string, changes) {
-	// No sorting needed as changes are added in correct order
+async function transform(code, changes, sourceMap) {
+	const mappingChanges = [];
 
-	const array = Array.from(string);
+	const array = Array.from(code);
+	// No sorting needed as changes are added in correct (reverse) order
 	changes.forEach((change) => {
+		if (sourceMap) {
+			// Compute line and column for given index to re-align source map with inserted characters
+			const precedingCode = array.slice(0, change.index);
+
+			const line = precedingCode.reduce((lineCount, char) => {
+				if (char === "\n") {
+					lineCount++;
+				}
+				return lineCount;
+			}, 0);
+			const lineStartIndex = precedingCode.lastIndexOf("\n") + 1;
+			const column = change.index - lineStartIndex;
+
+			// Source map re-alignment needs to be done from front to back
+			mappingChanges.unshift({
+				line,
+				column,
+				columnDiff: change.value.length
+			});
+		}
+
+		// Apply modification
 		array.splice(
 			change.index,
-			change.count,
+			0,
 			change.value
 		);
 	});
-	return array.join("");
+	const transformedCode = array.join("");
+
+	if (sourceMap) {
+		const mappings = decodeMappings(sourceMap.mappings);
+		mappingChanges.forEach((mappingChange) => {
+			const lineMapping = mappings[mappingChange.line];
+			if (!lineMapping) {
+				// No mapping available that could be transformed
+				return;
+			}
+			// Mapping structure:
+			// [generatedCodeColumn, sourceIndex, sourceCodeLine, sourceCodeColumn, nameIndex]
+			lineMapping.forEach((mapping) => {
+				if (mapping[0] > mappingChange.column) {
+					// All column mappings for the generated code after any change
+					// need to be moved by the amount of inserted characters
+					mapping[0] = mapping[0] + mappingChange.columnDiff;
+				}
+			});
+		});
+
+		sourceMap.mappings = encodeMappings(mappings);
+
+		// No need for file information in source map since the bundled code does not exist in any file anyways
+		delete sourceMap.file;
```

Comment:
Is there a reason why this is only removed in the "predefine" use case, not when written as sap.ui.require.preload?
As `createTransientSourceMap` also doesn't add a "file", I would prefer to also get rid of it for the preload use case.

Reply:
As discussed, the `file` attribute is optional and we don't really know what it's actually used for in current development tools. In this case however, there is actually no file that this source map belongs to, since it is a modified version of the source map of a minified resource.

In the case of transient source maps, they only belong to sections of a bundle resource. It might be incorrect to set the `file` attribute in that case, although I'm not sure about that.

In cases where we bundle whole resources without modifying their content, I think we should also take over their source maps without modification. I.e. without removing the `file` attribute.